### PR TITLE
[RECONSTRUCTION] fixes needed to build cxxmodules

### PR DESCRIPTION
**Pull request by smuzaffar**
#### PR description:

- Added missing dependencies in BuildFiles.
- Cleanup unused include statements.

```
DataFormats/CaloRecHit/interface/CaloCluster.h:#include "DataFormats/Math/interface/Point3D.h"

DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h:#include "DataFormats/EcalDigi/interface/EcalDataFrame.h"

DataFormats/EgammaReco/interface/BasicClusterFwd.h:#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
DataFormats/EgammaReco/interface/PreshowerClusterShapeFwd.h:#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
DataFormats/EgammaReco/interface/PreshowerClusterFwd.h:#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"

DataFormats/ParticleFlowCandidate/interface/PFCandidate.h:#include "DataFormats/MuonReco/interface/Muon.h"
DataFormats/ParticleFlowCandidate/interface/PFCandidate.h:#include "DataFormats/MuonReco/interface/MuonFwd.h"
DataFormats/ParticleFlowCandidate/src/PFCandidate.cc:#include "DataFormats/MuonReco/interface/Muon.h"

DataFormats/RecoCandidate/src/classes.h:#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
DataFormats/RecoCandidate/src/classes.h:#include "DataFormats/HcalRecHit/interface/HcalRecHitDefs.h" 
DataFormats/RecoCandidate/interface/TrackAssociation.h:#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
DataFormats/RecoCandidate/interface/TrackAssociation.h:#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticleFwd.h"

DataFormats/TrackReco/interface/DeDxHitInfo.h:#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"

DataFormats/TrackerRecHit2D/interface/ClusterRemovalRefSetter.h:#include "FWCore/Framework/interface/ConsumesCollector.h"
DataFormats/TrackerRecHit2D/interface/ClusterRemovalRefSetter.h:#include "FWCore/Framework/interface/Event.h"

DataFormats/TrackingRecHit/interface/TrackingRecHit.h:#include "DataFormats/GeometryCommonDetAlgo/interface/GlobalError.h"
DataFormats/TrackingRecHit/interface/AlignmentPositionError.h:#include "DataFormats/GeometryCommonDetAlgo/interface/GlobalError.h"
```

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: DataFormats/TrackerCommon/interface/ClusterSummary.h [hunk 1]
```diff
@@ -43,7 +43,7 @@
 #include "DataFormats/SiStripDigi/interface/SiStripProcessedRawDigi.h"
 #include "DataFormats/Common/interface/DetSetVector.h"
 
-#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+//#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
```
Comment (slava77) [inline DataFormats/TrackerCommon/interface/ClusterSummary.h]: please drop the commented out line.

**Comment by slava77** [comment]
@smuzaffar 
Please follow up or comment on
https://github.com/cms-sw/cmssw/pull/26958#pullrequestreview-242748530
Thank you.

File: DataFormats/TrackerCommon/interface/ClusterSummary.h (hunk 1 above)
Comment (smuzaffar) [reply DataFormats/TrackerCommon/interface/ClusterSummary.h]: done

**Comment by smuzaffar** [comment]
@slava77 , sorry for not paying attention to this. Commented line has been removed now.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/26958	029bfa3
- technical
- jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Pull request merged**